Library/Homebrew/formula_support.rb

```diff
@@ -28,13 +28,14 @@ def valid?
 
   def to_s
     return @explanation unless @explanation.empty?
+    os_name = OS::Mac.os_name(ENV["HOMEBREW_OSX_VERSION"])
```

Comment:
So I think when `test_formula_support` makes assertions on the user-facing messages, it should require `os/mac` otherwise it does not simulate the real cases.
Apparently we are facing same `os/mac` loading problem if we write a test for `KegOnlyReason::valid` method.